### PR TITLE
Make one env optional and warn about optional missing envs

### PR DESCRIPTION
**Pull request by nettnikl**
Fixes https://github.com/erikw/restic-automatic-backup-scheduler/issues/109

### CODE REVIEW

**Comment by nettnikl** [comment]
Have to do some testing still

File: bin/restic_backup.sh [hunk 1]
```diff
@@ -33,17 +33,36 @@ assert_envvars() {
 		fi
 	done
 }
+
+warn_on_missing_envvars() {
+	unset_envs=
```
Comment (erikw) [inline bin/restic_backup.sh:38]: Could make this variable `local`

Comment (nettnikl) [reply bin/restic_backup.sh]: Done, good point. Missing bash experience shows..

**Comment by nettnikl** [comment]
```sh
$ B2_ACCOUNT_KEY=xxx B2_CONNECTIONS=xxx RESTIC_PASSWORD_FILE= B2_ACCOUNT_ID= INSTALL_PREFIX= ./bin/restic_backup.sh
The following env variables are recommended, but have not been set. This script may not work as expected: B2_ACCOUNT_ID RESTIC_PASSWORD_FILE
reading repository password from stdin
Fatal: an empty password is not a password

$ B2_ACCOUNT_KEY=xxx B2_CONNECTIONS=xxx RESTIC_PASSWORD_FILE=xxx B2_ACCOUNT_ID= INSTALL_PREFIX= ./bin/restic_backup.sh
The following env variables are recommended, but have not been set. This script may not work as expected: B2_ACCOUNT_ID
Resolving password failed: Fatal: xxx does not exist

$ B2_ACCOUNT_KEY=xxx B2_CONNECTIONS=xxx RESTIC_PASSWORD_FILE=xxx B2_ACCOUNT_ID=1 INSTALL_PREFIX= ./bin/restic_backup.sh
Resolving password failed: Fatal: xxx does not exist
```

**Comment by nettnikl** [comment]
Hey @erikw, from my pov seems okay, if you are fine with the changes and you checked it doesn't break (even though works on my machine) you can merge!

**Comment by erikw** [comment]
Great, thank you!

**Pull request merged**